### PR TITLE
Update post and regional_workflow hashes, QOL improvements

### PR DESCRIPTION
**Pull request by mkavulich**
## DESCRIPTION OF CHANGES: 
EMC_post builds are currently failing on Hera due to an update to NCEPLIBS. While that is annoying, we also discovered that the App's EMC_post hash is from December 2020...way out of date! Updating to the latest hash fixes the build issue. In addition, regional_workflow must be updated and an additional PR must be applied (https://github.com/NOAA-EMC/regional_workflow/pull/581) in order to accommodate the new executable and filenames for the newer EMC_post hash.

In addition, some long-overdue quality-of-life improvements are added:
 - Including the "BUILD_ALWAYS" flag in CMakelists.txt; this will cause code to be properly rebuilt by cmake after making changes (already in the release branch from PR #91)
 - Adding "module load rocoto" to the Hera workflow environment file. Rocoto is not loaded by default and can be unloaded by the build environment, so adding this here (as it already exists for Cheyenne and Orion) makes sense.

## TESTS CONDUCTED: 
**Cheyenne**
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GSD_SAR
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
 - grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_2017_gfdlmp
 - grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_GSD_SAR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_HRRR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_RRFS_v1beta
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_GSD_SAR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_GSD_v0
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1alpha
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta

**Hera**
 - grid_CONUS_25km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_AK_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GSD_SAR
 - grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GSD_v0
 - grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16

## DEPENDENCIES:
 - https://github.com/NOAA-EMC/regional_workflow/pull/581

### CODE REVIEW

File: Externals.cfg [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [regional_workflow]
 protocol = git
-repo_url = https://github.com/NOAA-EMC/regional_workflow
+repo_url = https://github.com/JeffBeck-NOAA/regional_workflow
```
Comment (jwolff-ncar) [inline Externals.cfg]: Do you really want this set to Jeff's fork?

File: Externals.cfg [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 # Specify either a branch name or a hash but not both.
-#branch = develop
-hash = de1220b
+branch = bug/fix_post_filenames
```
Comment (jwolff-ncar) [inline Externals.cfg]: Not sure we want to use this branch either.

**Review by jwolff-ncar — COMMENTED** [review]
Couple of questions on the regional_workflow fork/branch set in this PR.

File: Externals.cfg (hunk 2 above)
Comment (mkavulich) [reply Externals.cfg]: This is pointing to the necessary fixes for testing convenience, I will update it when I convert this from a draft to an open PR.

Comment (jwolff-ncar) [reply Externals.cfg]: Got it - my apologies. I didn't realize this was a draft.

**Comment by mkavulich** [comment]
Ran a number of tests on Cheyenne. The following WE2E tests passed:

 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GSD_SAR
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
 - grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_2017_gfdlmp
 - grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_GSD_SAR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_HRRR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_HRRR_suite_RRFS_v1beta
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_GSD_SAR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_GSD_v0
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_HRRR
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1alpha
 - grid_RRFS_CONUS_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta

**Comment by mkavulich** [comment]
Ran another set of tests on Hera, all succeeded that were expected to (there are known failures in the testing system right now due to data being shuffled around). The following WE2E tests passed:
 - grid_CONUS_25km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_AK_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
 - grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GSD_SAR
 - grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GSD_v0
 - grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16

On Jet, I was not able to run any tests, as the old testing data has all been moved. I did build the code successfully.

**Comment by gsketefian** [comment]
@mkavulich Can you elaborate on what the BUILD_ALWAYS flag does?  What is the undesired behavior that this fixes?  Thanks.

**Comment by mkavulich** [comment]
@gsketefian The BUILD_ALWAYS flag ensures that when making changes to the source code in src/ and rebuilding, `cmake` will recognize that changes were made and `make` will rebuild with those changes as expected. This change was made in the release branch (#91) but never made it into develop.

**Review by JulieSchramm — APPROVED** [review]
Looks good.

**Review by gsketefian — APPROVED** [review]
Looks good.

**Pull request merged**